client/src/main/java/io/pravega/client/auth/CredentialsExtractor.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.stream.impl.Credentials;
+import java.util.Map;
+import java.util.Properties;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import javax.annotation.concurrent.GuardedBy;
+
+/**
+ * Helper class to extract credentials.
+ */
+public class CredentialsExtractor {
+    private static final String AUTH_PROPS_START = "pravega.client.auth.";
+    private static final String AUTH_METHOD = AUTH_PROPS_START + "method";
+    private static final String AUTH_METHOD_LOAD_DYNAMIC = AUTH_PROPS_START + "loadDynamic";
+
+    private static final String AUTH_PROPS_START_ENV = "pravega_client_auth_";
+
+    @GuardedBy("$this")
+    private final Properties properties;
+    @GuardedBy("$this")
+    private final Map<String, String> env;
+
+
+    public CredentialsExtractor() {
+        this(System.getProperties(), System.getenv());
+    }
+
+    @VisibleForTesting
+    CredentialsExtractor(Properties properties, Map<String, String> env) {
+        this.properties = properties;
+        this.env = env;
+    }
+
+    public static ClientConfig extractCreds(ClientConfig config) {
```

Comment:
I don't understand why we need a static method and a class method with the same signature. A more elegant way to do this would be to not have this class and to make it part of the `ClientConfig` builder to extract and add credentials. For example, we could have something like:

```
clientConfig = ClientConfig.builder()
                                             .fromConfig(clientConfig)
                                             .extractCredentials()
                                             .build();
```

I think this would be a lot more elegant and less prone to errors than having this class.

Also, in the case we need to maintain this method, let's put down the full name: `extractCredentials`.

Reply:
Fixed. Moved the logic to `ClientConfig`.